src/module.cpp

```diff
@@ -663,6 +663,71 @@ std::vector<shape> module::get_output_shapes() const
     }
 }
 
+std::vector<shape> module::compute_shapes(const std::vector<shape>& inputs,
+                                          compute_shapes_options options) const
+{
+    auto params = this->get_parameter_names();
+    std::sort(params.begin(), params.end());
```

Comment:
why is a sort needed here?

Reply:
The order of parameters when passing as arguments is to use alphabetic order.

Reply:
ah understood. Do we need the sort at all though? Why does alphabetical order matter for the arguments? Does the order matter here?

Reply:
> Do we need the sort at all though?

Yes of course because we pass them in alphabetic order to the submodule so we need to sort the names so they will be in alphabetic order. 

> Why does alphabetical order matter for the arguments?

That is the convention we have chosen for passing arguments to the submodule parameters.

> Does the order matter here?

Yes of course otherwise we wont know which argument goes with with parameter.